From 6a75428a51c1b8e444af579e1a9c901c8f0e2b0c Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:11:56 -0400
Subject: [ethernet] igb: specify phc_index of 82575 for get_ts_info

Message-id: <1409062359-13063-39-git-send-email-sassmann@redhat.com>
Patchwork-id: 89161
O-Subject: [RHEL7.1 PATCH 38/81] igb: specify phc_index of 82575 for get_ts_info
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Ken ICHIKAWA <ichikawa.ken@jp.fujitsu.com>

82575 has only software timestamping capability and it has
no PTP Hardware Clocks. Therefore, -1 has to be specified
to the phc_index for ethtool's get_ts_info, otherwise a wrong
value will be set to the phc_index.

v2: move the if (adapter->ptp_clock) section specifying phc_index
    to above the switch statement as suggested by Matthew Vick.
    adapter->ptpclock will always be NULL for 82575.

Signed-off-by: Ken ICHIKAWA <ichikawa.ken@jp.fujitsu.com>
Acked-by: Matthew Vick <matthew.vick@intel.com>
Tested-by: Jeff Pieper <jeffrey.e.pieper@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 0f49da0e78f445d7a7e3e566ae4871c9428b0448)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_ethtool.c b/drivers/net/ethernet/intel/igb/igb_ethtool.c
index d668815..33c289d 100644
--- a/drivers/net/ethernet/intel/igb/igb_ethtool.c
+++ b/drivers/net/ethernet/intel/igb/igb_ethtool.c
@@ -2353,6 +2353,11 @@ static int igb_get_ts_info(struct net_device *dev,
 {
  struct igb_adapter *adapter = netdev_priv(dev);
 
+ if (adapter->ptp_clock)
+  info->phc_index = ptp_clock_index(adapter->ptp_clock);
+ else
+  info->phc_index = -1;
+
  switch (adapter->hw.mac.type) {
  case e1000_82575:
   info->so_timestamping =
@@ -2374,11 +2379,6 @@ static int igb_get_ts_info(struct net_device *dev,
    SOF_TIMESTAMPING_RX_HARDWARE |
    SOF_TIMESTAMPING_RAW_HARDWARE;
 
-  if (adapter->ptp_clock)
-   info->phc_index = ptp_clock_index(adapter->ptp_clock);
-  else
-   info->phc_index = -1;
-
   info->tx_types =
    (1 << HWTSTAMP_TX_OFF) |
    (1 << HWTSTAMP_TX_ON);
-- 
1.7.1